### PR TITLE
Add arm/arm64 and chakrafull support to runtests.py

### PR DESCRIPTION
**Pull request by Penguinwizzard**
This allows running jshost on the core unit tests easily from the python runner, as well as running core tests on arm and arm64.
Note: hold this change, and re-target to release/1.10 when ready.

### CODE REVIEW

File: test/runtests.py [hunk 1]
```diff
@@ -94,7 +102,7 @@
     test_root = os.path.realpath(args.override_test_root)
 
 # arch: x86, x64
-arch = 'x86' if args.x86 else ('x64' if args.x64 else None)
+arch = 'x86' if args.x86 else ('x64' if args.x64 else ('arm' if args.arm else ('arm64' if args.arm64 else None)))
```
Comment (jackhorton) [inline test/runtests.py]: This... seems a bit ridiculous, especially with the two extra architectures. Could we make it into a dictionary lookup or a regular if tree?

File: test/runtests.py [hunk 2]
```diff
@@ -110,26 +118,46 @@
...
 # binary: full ch path
 binary = args.binary
 if binary == None:
     if sys.platform == 'win32':
         build = "VcBuild.SWB" if args.swb else "VcBuild"
-        binary = 'Build\\' + build + '\\bin\\{}_{}\\ch.exe'.format(arch, flavor)
+        binary = os.path.join('Build', build, 'bin', '{}_{}'.format(arch, flavor), binary_name)
     else:
-        binary = 'out/{0}/ch'.format(flavor)
+        binary = os.path.join('out', '{0}'.format(flavor), binary_name)
```
Comment (jackhorton) [inline test/runtests.py]: `'{0}'.format(flavor)` can just be `flavor`.

File: test/runtests.py [hunk 3]
```diff
@@ -506,7 +547,7 @@ def timeout_func(timeout_data):
             return self._show_failed(timedout=True, **fail_args)
 
         # check ch failed
-        if exit_code != 0:
+        if exit_code != 0 and binary_name_noext == 'ch':
```
Comment (jackhorton) [inline test/runtests.py]: Why would this be specific to ch?

Comment (Penguinwizzard) [reply test/runtests.py]: I believe that an uncaught exception in jshost causes us to exit with a non-zero error code, but in ch it causes us to exit with a 0 error code. We have some tests where the baseline includes the error message at the end. We may want to normalize this behavior in the future, but that's out-of-scope for this change.

File: test/runtests.py (hunk 1 above)
Comment (Penguinwizzard) [reply test/runtests.py]: I see your point, this is changed to a series of if statements in the next commit. I don't know of a good way to do this that's not significantly bigger or harder to read, though.

**Pull request merged**